OK, I just tested the `wangle` update from here on `staging-next`, and encountered:

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/k9hhrmnspybqbgc1h3ngzxl5kfsgyhwf-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LOCALEDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/share/doc/wangle -DCMAKE_INSTALL_INFODIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/bvlrfzqcnjixgyyzg9s9nqfjywg99ply-wangle-2024.12.09.00-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/bvlrfzqcnjixgyyzg9s9nqfjywg99ply-wangle-2024.12.09.00-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/strip -DCMAKE_RANLIB=/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/ranlib -DCMAKE_AR=/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00 -DBUILD_SHARED_LIBS:BOOL=TRUE -DBUILD_TESTS:BOOL=TRUE -DINCLUDE_INSTALL_DIR:STRING=/nix/store/bvlrfzqcnjixgyyzg9s9nqfjywg99ply-wangle-2024.12.09.00-dev/include -DLIB_INSTALL_DIR:STRING=/nix/store/5c0mnfh1f347b20xzqwg4q179fdvdjh5-wangle-2024.12.09.00/lib -DCMAKE_INSTALL_DIR:STRING=/nix/store/bvlrfzqcnjixgyyzg9s9nqfjywg99ply-wangle-2024.12.09.00-dev/lib/cmake/wangle
CMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The CXX compiler identification is GNU 14.2.1
-- The C compiler identification is GNU 14.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Warning (dev) at /nix/store/38cffsqqx823crf1i4bcf6zz1qz1hgpd-cmake-3.30.5/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  /nix/store/0c8pawm4gq7y84h5f5zn1v62273mfzza-folly-2024.11.18.00-dev/lib/cmake/folly/folly-config.cmake:59 (find_dependency)
  CMakeLists.txt:64 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boost: /nix/store/f0qbspim8mzycf2df88sqvvmfnd7yxxj-boost-1.86.0-dev/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found suitable version "1.86.0", minimum required is "1.51.0") found components: context filesystem program_options regex system thread
-- Found folly: /nix/store/0c8pawm4gq7y84h5f5zn1v62273mfzza-folly-2024.11.18.00-dev
-- Found Sodium: /nix/store/z6x0gkvc51068cyr1i2182klbsflbkjw-libsodium-1.0.20/lib/libsodium.so
-- Found folly: /nix/store/0c8pawm4gq7y84h5f5zn1v62273mfzza-folly-2024.11.18.00-dev
-- Found ZLIB: /nix/store/cqlaa2xf6lslnizyj9xqa8j0ii1yqw0x-zlib-1.3.1/lib/libz.so (found version "1.3.1")
-- Found fizz: /nix/store/lh0l3kwfm08ims446bywvyprm5nff3il-fizz-2024.11.18.00-dev
-- Found OpenSSL: /nix/store/h1ydpxkw9qhjdxjpic1pdc2nirggyy6f-openssl-3.3.2/lib/libcrypto.so (found version "3.3.2")
-- Found Glog: /nix/store/jx10wih0hr790svp4bnghmlhky35pyk0-glog-0.6.0/lib/libglog.so
-- Found gflags from package config /nix/store/4nh9giwncqvmmkl1mxs1vyv50lwyrxh5-gflags-2.2.2/lib/cmake/gflags/gflags-config.cmake
-- Found gflags as a dependency of glog::glog, include=/nix/store/4nh9giwncqvmmkl1mxs1vyv50lwyrxh5-gflags-2.2.2/include, libs=gflags_shared
CMake Warning (dev) at /nix/store/38cffsqqx823crf1i4bcf6zz1qz1hgpd-cmake-3.30.5/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
  The package name passed to `find_package_handle_standard_args` (LIBUNWIND)
  does not match the name of the calling package (LibUnwind).  This can lead
  to problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /build/source/build/fbcode_builder/CMake/FindLibUnwind.cmake:22 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  /build/source/build/fbcode_builder/CMake/FindGlog.cmake:44 (find_package)
  CMakeLists.txt:69 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find LIBUNWIND (missing: LIBUNWIND_LIBRARY LIBUNWIND_INCLUDE_DIR) 
-- Found gflags from package config
-- gflags_CONFIG=/nix/store/4nh9giwncqvmmkl1mxs1vyv50lwyrxh5-gflags-2.2.2/lib/cmake/gflags/gflags-config.cmake
-- Found libevent: /nix/store/pd23v45sv2f7ifcpj8cr0y45xq5vdrji-libevent-2.1.12/lib/libevent.so
-- Found DoubleConversion: /nix/store/v89la64ph3b9m09ccmfqxyi82ra465vl-double-conversion-3.3.0/lib/libdouble-conversion.so
-- Found Threads: TRUE
CMake Warning (dev) at /nix/store/38cffsqqx823crf1i4bcf6zz1qz1hgpd-cmake-3.30.5/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
  The package name passed to `find_package_handle_standard_args` (librt) does
  not match the name of the calling package (Librt).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/FindLibrt.cmake:26 (find_package_handle_standard_args)
  CMakeLists.txt:81 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found librt: /nix/store/65h17wjrrlsj2rj540igylrx7fqcd6vq-glibc-2.40-36/lib/librt.so
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB - Success
-- Looking for SSL_SESSION_dup in ssl
-- Looking for SSL_SESSION_dup in ssl - found
-- Found gmock via config, defines=, include=/nix/store/jjk5d24nfk0z87cg78cv5mzy2yj5z2fa-gtest-1.15.2-dev/include, libs=GTest::gmock_main;GTest::gmock;GTest::gtest
-- Configuring done (1.0s)
-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j1
[1/80] Building CXX object CMakeFiles/wangle.dir/acceptor/Acceptor.cpp.o
[2/80] Building CXX object CMakeFiles/wangle.dir/acceptor/AcceptorHandshakeManager.cpp.o
[3/80] Building CXX object CMakeFiles/wangle.dir/acceptor/ConnectionManager.cpp.o
[4/80] Building CXX object CMakeFiles/wangle.dir/acceptor/EvbHandshakeHelper.cpp.o
[5/80] Building CXX object CMakeFiles/wangle.dir/acceptor/FizzAcceptorHandshakeHelper.cpp.o
FAILED: CMakeFiles/wangle.dir/acceptor/FizzAcceptorHandshakeHelper.cpp.o 
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/g++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DFMT_SHARED -DGFLAGS_IS_A_DLL=0 -DWANGLE_HAVE_SSL_SESSION_DUP -Dwangle_EXPORTS -I/build/source/wangle/.. -isystem /nix/store/bbdwqad05z85sid6f74482xq9pacrc7p-fast-float-7.0.0/include -isystem /nix/store/26153nk3n2qm4ql1dkldzxcyb46r11as-xz-5.6.3-dev/include -isystem /nix/store/q17r2ngwxzkpdiyxpy8q2dj3c3h0nkj6-lz4-1.10.0-dev/include -isystem /nix/store/v4wsjzlrhzxra6dlzzwcpd6br55z10fd-zstd-1.5.6-dev/include -isystem /nix/store/kf1wnwjxd1j8s5wpglj0vnyjxf7b0n8v-libiberty-14-20241116-dev/include/libiberty -isystem /nix/store/jfl8wq4mm6srlxv2702hvjhw0v6cw0d1-libunwind-1.8.1-dev/include -O3 -DNDEBUG -std=c++17 -fPIC -MD -MT CMakeFiles/wangle.dir/acceptor/FizzAcceptorHandshakeHelper.cpp.o -MF CMakeFiles/wangle.dir/acceptor/FizzAcceptorHandshakeHelper.cpp.o.d -o CMakeFiles/wangle.dir/acceptor/FizzAcceptorHandshakeHelper.cpp.o -c /build/source/wangle/acceptor/FizzAcceptorHandshakeHelper.cpp
/build/source/wangle/acceptor/FizzAcceptorHandshakeHelper.cpp: In member function 'virtual void wangle::FizzAcceptorHandshakeHelper::handshakeSuc(folly::AsyncSSLSocket*)':
/build/source/wangle/acceptor/FizzAcceptorHandshakeHelper.cpp:259:49: error: 'struct fizz::server::AttemptVersionFallback' has no member named 'handshakeLogging'
  259 |   tinfo_.handshakeLogging = std::move(fallback_.handshakeLogging);
      |                                                 ^~~~~~~~~~~~~~~~
/build/source/wangle/acceptor/FizzAcceptorHandshakeHelper.cpp: In member function 'virtual void wangle::FizzAcceptorHandshakeHelper::handshakeErr(folly::AsyncSSLSocket*, const folly::AsyncSocketException&)':
/build/source/wangle/acceptor/FizzAcceptorHandshakeHelper.cpp:279:30: error: 'struct fizz::server::AttemptVersionFallback' has no member named 'handshakeLogging'
  279 |         *sock, ex, fallback_.handshakeLogging.get());
      |                              ^~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

--- a/pkgs/by-name/ed/edencommon/package.nix
+++ b/pkgs/by-name/ed/edencommon/package.nix
@@ -23,7 +23,7 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "edencommon";
-  version = "2024.11.18.00";
+  version = "2024.12.09.00";
 
   outputs = [
     "out"
@@ -34,7 +34,7 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "facebookexperimental";
     repo = "edencommon";
     rev = "refs/tags/v${finalAttrs.version}";
-    hash = "sha256-pVPkH80vowdpwWv/h6ovEk335OeI6/0k0cAFhhFqSDM=";
+    hash = "sha256-2kn7RCFNpsfyXFWhV7ikXiCCLMND185JRyxAnNq/1ro=";
   };
 
   patches = lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [

--- a/pkgs/by-name/fi/fizz/package.nix
+++ b/pkgs/by-name/fi/fizz/package.nix
@@ -28,7 +28,7 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "fizz";
-  version = "2024.11.18.00";
+  version = "2024.12.09.00";
 
   outputs = [
     "bin"
@@ -40,7 +40,7 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "facebookincubator";
     repo = "fizz";
     rev = "refs/tags/v${finalAttrs.version}";
-    hash = "sha256-mNe+CHEXhkwzek9qy2l6zvPXim9tJV44s+naSm6bQ4Q=";
+    hash = "sha256-fdaNIGndj3tCbMHE+nVtw+aTxrlaftz6gs9VFs1ANqs=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/by-name/fo/folly/char_traits.patch
+++ b/pkgs/by-name/fo/folly/char_traits.patch
@@ -1,0 +1,29 @@
+diff --git a/folly/memory/test/UninitializedMemoryHacksTest.cpp b/folly/memory/test/UninitializedMemoryHacksTest.cpp
+index 38e27c3..17424af 100644
+--- a/folly/memory/test/UninitializedMemoryHacksTest.cpp
++++ b/folly/memory/test/UninitializedMemoryHacksTest.cpp
+@@ -283,7 +283,7 @@ TEST(UninitializedMemoryHacks, simpleStringWChar) {
+ }
+ 
+ TEST(UninitializedMemoryHacks, simpleStringSChar) {
+-  testSimple<std::basic_string<signed char>>();
++  testSimple<std::basic_string<char>>();
+ }
+ 
+ TEST(UninitializedMemoryHacks, simpleVectorChar) {
+@@ -307,7 +307,7 @@ TEST(UninitializedMemoryHacks, randomStringWChar) {
+ }
+ 
+ TEST(UninitializedMemoryHacks, randomStringSChar) {
+-  testRandom<std::basic_string<signed char>>();
++  testRandom<std::basic_string<char>>();
+ }
+ 
+ TEST(UninitializedMemoryHacks, randomVectorChar) {
+@@ -323,5 +323,5 @@ TEST(UninitializedMemoryHacks, randomVectorInt) {
+ }
+ 
+ // We are deliberately putting this at the bottom to make sure it can follow use
+-FOLLY_DECLARE_STRING_RESIZE_WITHOUT_INIT(signed char)
++//FOLLY_DECLARE_STRING_RESIZE_WITHOUT_INIT(char)
+ FOLLY_DECLARE_VECTOR_RESIZE_WITHOUT_INIT(int)

--- a/pkgs/by-name/fo/folly/package.nix
+++ b/pkgs/by-name/fo/folly/package.nix
@@ -41,7 +41,7 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "folly";
-  version = "2024.11.18.00";
+  version = "2024.12.09.00";
 
   # split outputs to reduce downstream closure sizes
   outputs = [
@@ -53,7 +53,7 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "facebook";
     repo = "folly";
     rev = "refs/tags/v${finalAttrs.version}";
-    hash = "sha256-CX4YzNs64yeq/nDDaYfD5y8GKrxBueW4y275edPoS0c=";
+    hash = "sha256-uX63Zg0Dy5kHdDFYAX7kbVTXUyrWNsdz867VJ0191YI=";
   };
 
   nativeBuildInputs = [
@@ -124,6 +124,12 @@ stdenv.mkDerivation (finalAttrs: {
 
   doCheck = true;
 
+  patches = [
+    # The base template for std::char_traits has been removed in LLVM 19
+    # https://releases.llvm.org/19.1.0/projects/libcxx/docs/ReleaseNotes.html
+    ./char_traits.patch
+  ];
+
   # https://github.com/NixOS/nixpkgs/issues/144170
   postPatch = ''
     substituteInPlace CMake/libfolly.pc.in \
@@ -169,6 +175,8 @@ stdenv.mkDerivation (finalAttrs: {
           ]
           ++ lib.optionals stdenv.hostPlatform.isDarwin [
             "buffered_atomic_test.BufferedAtomic.singleThreadUnguardedAccess"
+            "io_async_notification_queue_test.NotificationQueueTest.UseAfterFork"
+            "container_heap_vector_types_test.HeapVectorTypes.SimpleSetTes"
           ]
         )
       )

